integration/v3_watch_test.go

```diff
@@ -1079,3 +1079,75 @@ func TestV3WatchWithFilter(t *testing.T) {
 		t.Fatal("failed to receive delete event")
 	}
 }
+
+func TestV3WatchWithPrevKV(t *testing.T) {
+	clus := NewClusterV3(t, &ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	tests := []struct {
+		key  string
+		end  string
+		vals []string
+	}{{
+		key:  "foo",
+		end:  "fop",
+		vals: []string{"bar1", "bar2"},
+	}, {
+		key:  "/abc",
+		end:  "/abd",
+		vals: []string{"first", "second"},
+	}}
+	for i, tt := range tests {
+		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
```

Comment:
we already use some ctx.TODO() in rpc calls with the assumption that the 1 node cluster wont die.

why do we need a ctx for the watch rpc?

we can change the ctx.done at line 1148 to a simple time.After.